### ISSUE TITLE
Should `with-init!` call its body immediately if the document has already loaded?

### ISSUE THREAD

**Issue by daveyarwood** [hoplon/hoplon#257]
I ran into some unexpected behavior re: `with-init!` 

It turns out that if you call `with-init!` _after_ the document has already loaded, the body will never be executed because it's too late at that point -- the document "load" event has already been fired and will not fire again.

I did some googling, and found [a good StackOverflow question](https://stackoverflow.com/questions/807878/javascript-that-executes-after-page-load) about how to handle this scenario in JavaScript. Multiple solutions mentioned checking to see if `document.readyState == "complete"`, so I tried that approach and it worked for my use case.

So here's my question: should this be the out-of-the-box behavior of `with-init!` ?

In my mind, `with-init!` should immediately execute its body if the document has already loaded; otherwise, it should register an event handler for "load" like it's doing now.

In other words:

```javascript
if (document.readyState == "complete") {
  f();
} else {
  window.addEventListener("load", f);
}
```

If the Hoplon maintainers/community are OK with making this change, I'm happy to submit a PR.

**Comment by flyboarder** [comment]
Open to the change, please put a PR.
